grammar.js

```diff
@@ -395,7 +398,105 @@ module.exports = grammar({
 
     comclear_statement: ($) => tokalias($, "comclear"),
 
+    // :h :command
     command_name: ($) => /[A-Z][A-Za-z0-9]*/,
+
+    _command_attribute_completion_behavior: ($) =>
+      choice(
+        ...[
+          'arglist',
+          'augroup',
+          'buffer',
+          'behave',
+          'color',
+          'command',
+          'compiler',
+          'cscope',
+          'dir',
+          'environment',
+          'even',
+          'expression',
+          'file',
+          'file_in_path',
+          'filetype',
+          'function',
+          'help',
+          'highlight',
+          'history',
+          'local',
+          'lua',
+          'mapclear',
+          'mapping',
+          'menu',
+          'messages',
+          'option',
+          'packadd',
+          'shellcmd',
+          'sign',
+          'syntax',
+          'syntime',
+          'tag',
+          'tag_listfiles',
+          'user',
+          'var',
+        ].map(token.immediate),
+        seq(token.immediate('custom'), token.immediate(','), $._ident),
+        seq(token.immediate('customlist'), token.immediate(','), $._ident),
+      ),
+    _command_attribute_address_behavior: ($) =>
+      choice(
+        ...[
+          'lines',
+          'arguments',
+          'buffers',
+          'loaded_buffers',
+          'windows',
+          'tabs',
+          'quickfix',
+          'other',
+        ].map(token.immediate),
+      ),
+
+    _command_attribute_nargs_value: ($) =>
+      choice(
+        alias(token.immediate(/[01]/), $.integer_literal),
+        alias(token.immediate(/[*?+]/), $.pattern_multi),
+      ),
+    _command_attribute_range_value: ($) =>
+      choice(
+        alias(token.immediate(/[0-9]+/), $.integer_literal),
+        alias(token.immediate(/%/), $.pattern_multi),
+      ),
+
+    command_attribute: ($) =>
+      choice(
+        seq(field('name', '-nargs'), token.immediate('='), field('value', $._command_attribute_nargs_value)),
+        seq(field('name', '-complete'), token.immediate('='), field('value', alias($._command_attribute_completion_behavior, $.behavior))),
+        seq(field('name', '-range'), optional(seq(token.immediate('='), field('value', $._command_attribute_range_value)))),
+        seq(field('name', '-count'), token.immediate('='), field('value', alias(token.immediate(/[0-9]+/), $.integer_literal))),
+        seq(field('name', '-addr'), token.immediate('='), field('value', alias($._command_attribute_address_behavior, $.behavior))),
+        field('name', '-bang'),
+        field('name', '-bar'),
+        field('name', '-register'),
+        field('name', '-buffer'),
+        field('name', '-keepscript'),
```

Comment:
Could be worth using one of the functions we use for `syntax` here ?

we could use `key_val_arg` ?